### ISSUE TITLE
SyntaxError: Type mismatch in unrelated `RETURN` column

### ISSUE THREAD

**Issue by CelineWuest** [neo4j/neo4j#13164]
When running the following query:
```cypher
RETURN null, [avg(0), endNode(null)] ORDER BY (CASE WHEN (true OR null) THEN 0 END)
```

I encountered the following error:
```
Neo4jError: Neo.ClientError.Statement.SyntaxError (Type mismatch: expected Boolean but was Relationship (line 1, column 8 (offset: 7))
"RETURN null, [avg(0), endNode(null)] ORDER BY (CASE WHEN (true OR null) THEN 0 END)"
        ^)
```

I believe the query mentioned above is semantically and syntactically correct and thus no error should be thrown here.

I encountered this issue when testing queries against the **Neo4j 5.6.0 enterprise version** in a Docker container running **alpine v.3**.

### Steps to reproduce
Run the following query and observe it throws an error:
```cypher
RETURN null, [avg(0), endNode(null)] ORDER BY (CASE WHEN (true OR null) THEN 0 END)
```

### Expected behavior
The query should run successfully

### Actual behavior
The query fails with the error message `Neo4jError: Neo.ClientError.Statement.SyntaxError (Type mismatch: expected Boolean but was Relationship [..]`

**Comment by loveleif** [comment]
Thanks @DominicWuest! I'm not completely convinced that the query is semantically correct (the `CASE WHEN (true OR null)` part) but might be wrong. However, the error message is really confusing. I've created a ticket for our internal bug tracker for further investigation ([link](https://trello.com/c/0SYGfFjf)).

**Comment by Lojjs** [comment]
@DominicWuest This query is semantically correct (although maybe not super useful in practice...). We found the same bug internally recently and it has been fixed in 5.7.0 that was released yesterday.
